checkstyle.xml

```diff
@@ -0,0 +1,122 @@
+        <module name="UnusedImports"/>
+
+        <!-- ##### Javadocs requirements ##### -->
+        <!-- Requirements for Javadocs for classes/interfaces -->
+        <module name="JavadocType">
+            <!-- All public classes/interfaces MUST HAVE Javadocs -->
+            <property name="scope" value="public"/>
+            <!-- Add an exception for anonymous inner classes -->
+            <property name="excludeScope" value="anoninner"/>
+            <!-- Ignore errors related to unknown tags -->
+            <property name="allowUnknownTags" value="true"/>
+            <!-- Allow params tags to be optional -->
+            <property name="allowMissingParamTags" value="false"/>
+        </module>
+        <!-- Requirements for Javadocs for methods -->
+        <module name="JavadocMethod">
+            <!-- All public methods MUST HAVE Javadocs -->
+            <property name="scope" value="public"/>
+            <!-- Allow RuntimeExceptions to be undeclared -->
+            <property name="allowUndeclaredRTE" value="true"/>
+            <!-- Allow params, throws and return tags to be optional -->
+            <property name="allowMissingParamTags" value="true"/>
+            <property name="allowMissingThrowsTags" value="true"/>
+            <property name="allowMissingReturnTag" value="true"/>
+        </module>
+
+        <!-- ##### Requirements for K&R Style braces ##### -->
+        <!-- Code blocks MUST HAVE braces, even single line statements (if, while, etc) -->
+        <module name="NeedBraces"/>
+        <!-- Left braces should be at the end of current line (default value)-->
+        <module name="LeftCurly"/>
+        <!-- Right braces should be on start of a new line (default value) -->
+        <module name="RightCurly"/>
+
+        <!-- ##### Indentation / Whitespace requirements ##### -->
+        <!-- Require 4-space indentation (default value), except "throws" clauses should be indented 8 spaces -->
+        <module name="Indentation">
+            <property name="throwsIndent" value="8"/>
+        </module>
+        <!-- Whitespace should exist around all major tokens -->
+        <module name="WhitespaceAround">
+            <!-- However, make an exception for empty constructors, methods, types, etc. -->
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+            <property name="allowEmptyTypes" value="true"/>
+            <property name="allowEmptyLoops" value="true"/>
+        </module>
+        <!-- Validate whitespace around Generics (angle brackets) per typical conventions
+             http://checkstyle.sourceforge.net/config_whitespace.html#GenericWhitespace -->
+        <module name="GenericWhitespace"/>
+
+        <!-- ##### Other / Miscellaneous requirements ##### -->
+        <!-- Require utility classes do not have a public constructor -->
+        <module name="HideUtilityClassConstructor"/>
```

Comment:
Related to visibility, I propose to also disallow direct class member access except for protected and final constant fields:

```
<module name="VisibilityModifier">
    <property name="protectedAllowed" value="true"/>
    <property name="allowPublicFinalFields" value="true"/>
    <property name="allowPublicImmutableFields" value="true"/>
</module>
```

http://checkstyle.sourceforge.net/config_design.html#VisibilityModifier

In addition, I would also prevent any redundant visibility modifiers to keep to code clean:
`<module name="RedundantModifier"/>` -> http://checkstyle.sourceforge.net/config_modifier.html#RedundantModifier